src/indicators/drought/wrsi.py

```diff
@@ -0,0 +1,309 @@
+    to around 17, and from the eastern border of Chad
+    to Senegal. This is specifically for rangelands:
+    https://earlywarning.usgs.gov/fews/product/57
+
+Both are published using ESRI:102022. There is overlap between the
+two datasets where they don't agree, even though both are
+measuring the WRSI with respect to millet, due to their
+calculations for rangeland and cropland. Since most
+countries in the Sahel overlap with both datasets, both
+should be considered and we should get clarification from
+experts on the differences between the two, how they relate,
+and how best to use them.
+
+I have also just found the CHIRPS WRSI data on the USGS website.
+These are available at:
+
+https://edcftp.cr.usgs.gov/project/fews/africa/west/dekadal/wrsi-chirps-etos
+"""
+import datetime
+import logging
+import os
+from datetime import date
+from io import BytesIO
+from pathlib import Path
+from typing import List, Literal, Union, get_args
+from urllib.error import HTTPError
+from urllib.request import urlopen
+from zipfile import ZipFile
+
+import numpy as np
+import pandas as pd
+import rioxarray  # noqa: F401
+import xarray as xr
+from rasterio.crs import CRS
+
+logger = logging.getLogger(__name__)
+
+_BASE_URL = (
+    "https://edcftp.cr.usgs.gov/project/fews/"
+    "africa/west/dekadal/wrsi-chirps-etos/{region}/"
+)
+
+_URL_ZIP_FILENAME = "w{year:04}{dekad}{region}.zip"
+
+_BASE_FILENAME = "w{year:04}{dekad}{type}.tif"
+
+
+def _get_url_path(year, dekad, region):
+    if region == "cropland":
+        url_region, file_region = "west", "wa"
+    else:
+        url_region, file_region = "west1", "w1"
+
+    url_zip_filename = _URL_ZIP_FILENAME.format(
+        year=year, dekad=dekad, region=file_region
+    )
+
+    base_url = _BASE_URL.format(region=url_region)
+    return base_url + url_zip_filename
+
+
+def _download_wrsi(year: int, dekad: int, region: str, raw_dir: Path):
+    try:
+        url = _get_url_path(year=year, dekad=dekad, region=region)
+        resp = urlopen(url)
+    except HTTPError:
+        logger.error(
+            f"No data available for region {region} for "
+            f"dekad {dekad} of {year}, skipping."
+        )
+        return
+
+    zf = ZipFile(BytesIO(resp.read()))
+    for type in ["do", "eo", "er"]:
+        file_name = _BASE_FILENAME.format(year=year, dekad=dekad, type=type)
+        save_path = Path(raw_dir, file_name)
+        save_path.unlink(missing_ok=True)
+
+        zf.extract(file_name, raw_dir)
+
+
+_RAW_PATH = Path(
+    os.getenv("AA_DATA_DIR"), "public", "raw", "glb", "wrsi", "west_africa"
+)
+
+
+def _get_raw_dir(region):
+    return Path(_RAW_PATH, region)
+
+
+def _date_to_dekad(date_obj: Union[date, str]):
+    if isinstance(date_obj, str):
+        date_obj = date.fromisoformat(date_obj)
+    year = date_obj.year
+    dekad = (
+        (date_obj.timetuple().tm_mday // 10) + ((date_obj.month - 1) * 3) + 1
+    )
+    return year, dekad
+
+
+def _dekad_to_date(year: int, dekad: int):
```

Comment:
even though private function I would add short docstring if the computed date represents the beginning of the dekad (would assume so) or e.g. the end

Reply:
Yes, didn't document this well yet, but going through now!